### PR TITLE
Aveybranch

### PR DESCRIPTION
**Pull request by dspages**
Adds a new action type, ActionBullRush, which has the following properties:

-Must end in an attack
-Gets four bonus movement points to move toward the target
-If it hits, it causes the target to be knocked down for one turn (effectively meaning it loses next turn)

### CODE REVIEW

File: Assets/Scripts/Combat/CombatController.cs [hunk 1]
```diff
@@ -40,7 +54,7 @@ public void BeginTurn()
         }
         isTurn = true;
         AssignCurrentTile();
```
Comment (basile-laderchi) [inline Assets/Scripts/Combat/CombatController.cs]: I would get rid of this line (it's already incorporated in FindSelectableBasicTiles).

File: Assets/Scripts/Combat/CombatController.cs [hunk 2]
```diff
@@ -71,13 +85,40 @@ private void AttachTile(int moveCostOverride, Tile adjacentTile, Tile parent)
         }
     }
 
-    protected void FindSelectableTiles()
+    // Returns true if valid charge tiles were found.
+    protected bool FindSelectableChargeTiles()
+    {
+        ClearVisitedTiles();
```
Comment (basile-laderchi) [inline Assets/Scripts/Combat/CombatController.cs]: I would incorporate this as 1st line in the FindSelectableTiles method and get rid of it from here.

File: Assets/Scripts/Combat/CombatController.cs [hunk 3]
```diff
@@ -71,13 +85,40 @@ private void AttachTile(int moveCostOverride, Tile adjacentTile, Tile parent)
...
+        {
+            FindSelectableBasicTiles();
+            return false;
+        }
+        return true;
+    }
+
+    protected void FindSelectableBasicTiles()
+    {
+        ClearVisitedTiles();
```
Comment (basile-laderchi) [inline Assets/Scripts/Combat/CombatController.cs]: Same comment as per line 91.

File: Assets/Scripts/Combat/CombatController.cs [hunk 4]
```diff
@@ -164,7 +211,7 @@ public void EndAction(int spentActionPoints)
             isTurn = false;
             return;
         }
-        FindSelectableTiles();
+        FindSelectableBasicTiles();
```
Comment (basile-laderchi) [inline Assets/Scripts/Combat/CombatController.cs:212]: I would get rid of the ClearVisitedTiles (7 lines above) and incorporate it as 1st line in the FindSelectableTiles method.

**Review by basile-laderchi — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Combat/CombatController.cs (hunk 1 above)
Comment (dspages) [reply Assets/Scripts/Combat/CombatController.cs]: Good point, it's redundant.

File: Assets/Scripts/Combat/CombatController.cs (hunk 3 above)
Comment (dspages) [reply Assets/Scripts/Combat/CombatController.cs]: Done

File: Assets/Scripts/Combat/CombatController.cs (hunk 2 above)
Comment (dspages) [reply Assets/Scripts/Combat/CombatController.cs]: Done

**Pull request merged**